apps/dashboard/src/components/layouts/layout-list.tsx

```diff
@@ -0,0 +1,198 @@
+import { HTMLAttributes, useEffect, useState } from 'react';
+import { RiAddCircleLine } from 'react-icons/ri';
+import { DirectionEnum, PermissionsEnum } from '@novu/shared';
+
+import { cn } from '@/utils/ui';
+import { CursorPagination } from '@/components/cursor-pagination';
+import { Table, TableBody, TableHead, TableHeader, TableRow } from '@/components/primitives/table';
+import {
+  LayoutsFilter,
+  LayoutsSortableColumn,
+  LayoutsUrlState,
+  useLayoutsUrlState,
+} from '@/components/layouts/hooks/use-layouts-url-state';
+import { LayoutListBlank } from '@/components/layouts/layout-list-blank';
+import { LayoutListNoResults } from '@/components/layouts/layout-list-no-results';
+import { LayoutRow, LayoutRowSkeleton } from '@/components/layouts/layout-row';
+import { LayoutsFilters } from '@/components/layouts/layouts-filters';
+import { useFetchLayouts } from '@/hooks/use-fetch-layouts';
+import { PermissionButton } from '@/components/primitives/permission-button';
+
+type LayoutListFiltersProps = HTMLAttributes<HTMLDivElement> &
+  Pick<LayoutsUrlState, 'filterValues' | 'handleFiltersChange' | 'resetFilters'> & {
+    isFetching?: boolean;
+  };
+
+const LayoutListWrapper = (props: LayoutListFiltersProps) => {
+  const { className, children, filterValues, handleFiltersChange, resetFilters, isFetching, ...rest } = props;
+
+  return (
+    <div className={cn('flex h-full flex-col p-2', className)} {...rest}>
+      <div className="flex items-center justify-between">
+        <LayoutsFilters
+          onFiltersChange={handleFiltersChange}
+          filterValues={filterValues}
+          onReset={resetFilters}
+          isFetching={isFetching}
+          className="py-2.5"
+        />
+        <PermissionButton
+          permission={PermissionsEnum.WORKFLOW_WRITE}
```

Comment:
only users with write permission can create layouts; similar applies to delete, duplicate